src/main/java/org/snomed/snowstorm/core/data/services/SBranchService.java

```diff
@@ -194,4 +196,15 @@ public Date getPartialCommitTimestamp(String branchPath) {
 		}
 		return null;
 	}
+
+	public Branch setAuthorFlag(String branchPath, SetAuthorFlag setAuthorFlag) {
+		Branch branch = branchService.findBranchOrThrow(branchPath);
+
+		Metadata metadata = branch.getMetadata();
+		Map<String, String> authFlagMap = metadata.getMapOrCreate(AUTHOR_FLAGS_METADATA_KEY);
+		authFlagMap.put(setAuthorFlag.getName(), String.valueOf(setAuthorFlag.isValue()));
+		metadata.putMap(AUTHOR_FLAGS_METADATA_KEY, authFlagMap);
+
+		return branchService.updateMetadata(branchPath, metadata);
+	}
```

Comment:
This logic was previously in `BranchController`, however I needed it in another place (`ConceptService`). 

I originally tried injecting the controller to the service, but that caused a circular dependency so I moved the logic to a separate service class.